sigar_linux_common.go

```diff
@@ -55,20 +55,26 @@ func (self *LoadAverage) Get() error {
 func (self *Mem) Get() error {
 	var buffers, cached uint64
 	table := map[string]*uint64{
-		"MemTotal": &self.Total,
-		"MemFree":  &self.Free,
-		"Buffers":  &buffers,
-		"Cached":   &cached,
+		"MemTotal":     &self.Total,
+		"MemFree":      &self.Free,
+		"MemAvailable": &self.ActualFree,
+		"Buffers":      &buffers,
+		"Cached":       &cached,
 	}
 
 	if err := parseMeminfo(table); err != nil {
 		return err
 	}
 
-	self.Used = self.Total - self.Free
 	kern := buffers + cached
-	self.ActualFree = self.Free + kern
-	self.ActualUsed = self.Used - kern
+	if self.ActualFree > 0.0 {
```

Comment:
You should be able to unit test this pretty easily by simulating the contents of the `/proc/meminfo` file.

With the current data structure it might not be possible to detect the difference between `MemAvailable: 0 kB` and `MemAvailable` not present. You don't want to fallback to the secondary calculation when `MemAvailable: 0 kB` so you might need to refactor.

`if self.ActualFree > 0.0 {` doesn't look right?